jsx/Tabs.js

```diff
@@ -141,6 +141,123 @@ Tabs.defaultProps = {
   updateURL: false
 };
 
+/**
+ * Allows to dynamically render vertical tabs corresponding to tab panes.
+ */
+
+class VerticalTabs extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const hash = window.location.hash;
+    let activeTab = "";
+
+    /**
+     * Determine the initial active tab in this order
+     * 1. Try to infer from the URL, otherwise
+     * 2. Try to infer from the defaultTab prop, otherwise
+     * 3. Set to be the first tab of the list
+     */
+    if (this.props.updateURL && hash) {
+      activeTab = hash.substr(1);
+    } else if (this.props.defaultTab) {
+      activeTab = this.props.defaultTab;
+    } else if (this.props.tabs.length > 0) {
+      activeTab = this.props.tabs[0].id;
+    }
+
+    this.state = {
+      activeTab: activeTab
+    };
+
+    this.handleClick = this.handleClick.bind(this);
+    this.getTabs = this.getTabs.bind(this);
+    this.getTabPanes = this.getTabPanes.bind(this);
+  }
+
+  handleClick(tabId, e) {
+    this.setState({activeTab: tabId});
+    this.props.onTabChange(tabId);
+
+    // Add tab href to URL querystring and scroll the page to top
+    if (this.props.updateURL) {
+      const scrollDistance = $('body').scrollTop() || $('html').scrollTop();
+      window.location.hash = e.target.hash;
+      $('html,body').scrollTop(scrollDistance);
+    }
+  }
+
+  getTabs() {
+    let tabs = (this.props.tabs).map(function(tab) {
+      let tabClass = this.state.activeTab === tab.id ? 'active' : null;
+      let href = "#" + tab.id;
+      let tabID = "tab-" + tab.id;
+      return (
+        <li
+          role="presentation"
+          className={tabClass}
+          key={tab.id}
+        >
+          <a id={tabID}
+             href={href}
+             role="tab"
+             data-toggle="tab"
+             onClick={this.handleClick.bind(null, tab.id)}
+          >
+            {tab.label}
+          </a>
+        </li>
+      );
+    }.bind(this));
+
+    return tabs;
+  }
+
+  getTabPanes() {
+    let tabPanes = React.Children.map(this.props.children, function(child, key) {
+      if (child) {
+        return React.cloneElement(child, {
```

Comment:
Why does this need to be cloned?

Reply:
copy of line 101 - regular tabs

not sure why, i could investigate

Reply:
It just smells like somewhere that shouldn't be is mutating things